packages/url_launcher/url_launcher_web/lib/src/link.dart

```diff
@@ -113,14 +113,28 @@ class LinkViewController extends PlatformViewController {
   /// platform view [params].
   factory LinkViewController.fromParams(
     PlatformViewCreationParams params,
-    BuildContext context,
-  ) {
-    final int viewId = params.id;
-    final LinkViewController controller = LinkViewController(viewId, context);
-    controller._initialize().then((_) {
+  ) =>
+      LinkViewController(params.id).._asyncInitialize(params);
+
+  Future<void> _asyncInitialize(PlatformViewCreationParams params) async {
+    try {
+      await SystemChannels.platform_views
+          .invokeMethod<void>('create', <String, dynamic>{
+        'id': viewId,
+        'viewType': linkViewType,
+      });
+      if (_isDisposed) {
+        return;
```

Comment:
I think the whole point of this fix is to check _after_ the await, sort of like you do when do `if (!mounted) return;` after awaits in stateful widget methods. Besides, I can't see how `_isDisposed` can be true before the await.

Reply:
> I can't see how `_isDisposed` can be true before the await.

@kristoffer-zliide the lifecycle of Flutter Widgets and platform views is somewhat separate. In this code, you're adding elements to the DOM ([code](https://github.com/flutter/engine/blob/7f2af2c5ec64752b1d133eab8016eeaf8c3c65d8/lib/web_ui/lib/src/engine/platform_dispatcher.dart#L483-L492)) and then checking if the Widget has been disposed to then *not* call onPlatformViewCreated.

I don't think you need to modify the DOM at all, or call "create" on a "platform_view" if the Widget that is being initialized has been previously marked as disposed. That's why I asked you to move the `isDisposed` check above the `invokeMethod`, **to exit as early as possible.**

In fact, if `onPlatformViewCreated` could be called multiple times earlier, [this should be complaining about `viewId` being already created](https://github.com/flutter/engine/blob/1b9fb67c428f67be7900012464ed3047d642efca/lib/web_ui/lib/src/engine/platform_views/message_handler.dart#L84-L91). (Is the `reportError` swallowing the exception on the web? Looking.)

_(Note: this LinkViewController was originally modeled to look like the [`class _HtmlElementViewController`, here](https://github.com/flutter/flutter/blob/64db62198018240ac517046af8714804841a863f/packages/flutter/lib/src/widgets/platform_view.dart#L386), so everything used `initialized` instead of `disposed`)_

Reply:
I appreciate the desire to skip anything that would need to be awaited, but I don't quite follow anything else you're writing. With the refactoring done in this PR, I don't think I need to read more than 15 lines of code in the Link.dart file to see that the check wouldn't do anything before the await, since `_asyncInitialize` is only called on newly constructed `LinkViewController`s which have `_isDisposed` set to false. *Moving* the check up also implies not doing it after, which was the whole point of the PR. But maybe I missed something in the refactoring?

Reply:
I'm just trying to understand what's flutter doing here, because I don't think we should be getting an "init" on something that has been already "disposed".

Reply:
The framework seems to be disposing of the link while the `await invokeMethod 'create'` is executing, so when it's time to call `onPlatformViewCreated`, `viewId` is already disposed.

Something like:

```
Disposed? false - before await create 428 - 918280561
!!! - dispose 428 - 918280561
Disposed? true - after await create 428 - 918280561
Exception! Kaboom 428 - 918280561
```